### PR TITLE
#167334075 User should be able to delete, update and mark as sold his own product

### PR DESCRIPTION
**Pull request by Ally4**
#### What does this PR do?
The PR is to allow the user to:
-Update,
-Delete,
-Mark as sold;
but it does so only for the owner of the project
#### Description of Task to be completed?
We look for a way of reading the token by using the JSON web toke verify, the after having that done, we will take that token and put it in the header for restriction in order for the person to be the only one to dele, update or mark as sold.  
#### How should this be manually tested?
You open the terminal
You run the npm start
You use postman by signing up
Then you sign in
The post a property
Then take your token and put it in the header on any endpoint you want to work on the start the changes.
#### Any background context you want to provide?
No, any other user can delete, update or even mark as sold what he/she did not create
#### What are the relevant pivotal tracker stories?
#167334075

### CODE REVIEW

File: server/controllers/sold.js [hunk 1]
```diff
@@ -11,25 +11,47 @@
 /* eslint-disable import/no-unresolved */
```
Comment (b0nbon1) [inline server/controllers/sold.js:11]: You have a lot commented out eslint errors, whats up?

**Review by b0nbon1 — COMMENTED** [review]
you can merge though check on your eslint comments, you can rules on eslintrc file to configure it. to reduce the comments

File: server/controllers/sold.js (hunk 1 above)
Comment (Ally4) [reply server/controllers/sold.js:11]: yes I will work on that too

File: server/controllers/createPosts.js [hunk 2]
```diff
@@ -1,35 +1,51 @@
+/* eslint-disable linebreak-style */
 /* eslint-disable quotes */
-/* eslint-disable comma-spacing */
-/* eslint-disable key-spacing */
+/* eslint-disable import/order */
+/* eslint-disable padded-blocks */
 /* eslint-disable indent */
```
Comment (Kasulejoseph) [inline server/controllers/createPosts.js:5]: Configure eslint in a eslint file.

File: server/models/posts.js [hunk 3]
```diff
@@ -1,100 +1,25 @@
...
+    status:'Available',
+    type:'2room',
+    state:'jhjhhgh',
+    city:'Kigali',
+    address:'fgghhgb',
+    price:1000,
+},
+{
+    id: 2,
+    owner:'su@gmail.com',
```
Comment (Kasulejoseph) [inline server/models/posts.js:14]: Remove this hardcoded data from models.

File: server/models/users.js [hunk 4]
```diff
@@ -1,55 +1,25 @@
...
     id: 1,
-    email: 'admin@gmail.com',
-    firstName: 'Ally',
-    lastName: 'NENGO',
-    password, 
-    address: 'Rwanda',
+    email: 'sunny1@gmail.com',
+    firstName: 'AGAHA',
+    lastName: 'baba',
+    password: '$2b$10$cpUWwMMkF8pJTxuNkT1Pf.CLhc/bmae18YSGnUp8rvlxM1BT025Jq',
```
Comment (Kasulejoseph) [inline server/models/users.js:8]: Remove the hardcoded data.

**Pull request merged**